sale_order_currency_rate/models/sale_order.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2023 Jarsa
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    currency_rate = fields.Float(
+        compute="_compute_currency_rate",
+        string="Currency",
+        readonly=True,
+        store=True,
+    )
+
+    inverse_currency_rate = fields.Float(
+        compute="_compute_currency_rate",
+        string="Inverse Currency",
+        readonly=True,
+        store=True,
+        digits=(12, 6),
```

Comment:
```suggestion
        digits="Currency Rate Precision"
```

And create a digital.precision record with data